build.proj

```diff
@@ -2,61 +2,88 @@
+
+    <BuildDependsOn Condition="'$(BuildRestore)'=='true'">$(BuildDependsOn);Restore</BuildDependsOn>
+    <BuildDependsOn Condition="'$(BuildNative)'=='true'">$(BuildDependsOn);BuildNative</BuildDependsOn>
+    <BuildDependsOn>$(BuildDependsOn);BuildManaged</BuildDependsOn>
+    <BuildDependsOn Condition="'$(BuildTests)'=='true'">$(BuildDependsOn);Test</BuildDependsOn>
+    <BuildDependsOn Condition="'$(BuildPackages)'=='true'">$(BuildDependsOn);Pack</BuildDependsOn>
+
+    <!-- Explicitly set Configuration based on BuildConfiguration for the root projects -->
+    <ProjectProperties>Configuration=$(BuildConfiguration)</ProjectProperties>
   </PropertyGroup>
 
   <Import Project="$(ToolsDir)VersionTools.targets" Condition="Exists('$(ToolsDir)VersionTools.targets')" />
 
-  <ItemGroup>
-    <Project Include="external\dir.proj" />
-    <Project Include="src\dirs.proj" />
-    <Project Include="src\tests.builds" Condition="'$(BuildTests)'=='true'" />
-    <!-- signing must happen before packaging -->
-    <Project Include="src\sign.proj" />
-    <Project Include="src\packages.builds" Condition="'$(BuildPackages)'=='true'" />
-  </ItemGroup>
-
   <Import Project="Directory.Build.targets" />
 
-  <Import Project="dir.traversal.targets" />
+  <UsingTask TaskName="GenerateConfigurationProps" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
 
-  <Import Project="$(ToolsDir)clean.targets" />
+  <Target Name="Restore" DependsOnTargets="CreateOrUpdateCurrentVersionFile">
+    <ItemGroup>
+      <_RestoreProjects Include="external\dir.proj" />
+    </ItemGroup>
 
-  <PropertyGroup>
-    <PrebuildSetupTargets>
-      GenerateConfigurationProperties;
-      CreateOrUpdateCurrentVersionFile
-    </PrebuildSetupTargets>
-  </PropertyGroup>
+    <!-- Generate the configurations for this repo -->
+    <GenerateConfigurationProps Properties="@(Property)" PropertyValues="@(PropertyValue)" PropsFolder="$(BuildConfigurationFolder)" />
 
-  <PropertyGroup>
-    <TraversalBuildDependsOn>
-      $(PrebuildSetupTargets);
-      $(TraversalBuildDependsOn);
-    </TraversalBuildDependsOn>
-  </PropertyGroup>
+    <MSBuild Projects="@(_RestoreProjects)" Properties="$(ProjectProperties)" />
+  </Target>
 
-  <!-- Generate the configurations for this repo -->
-  <UsingTask TaskName="GenerateConfigurationProps" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
-  <Target Name="GenerateConfigurationProperties">
-    <GenerateConfigurationProps Properties="@(Property)" PropertyValues="@(PropertyValue)" PropsFolder="$(BuildConfigurationFolder)" />
+  <Target Name="BuildNative">
+    <ItemGroup>
+      <_BuildNativeProjects Include="src\Native\build-native.proj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(_BuildNativeProjects)" Properties="$(ProjectProperties)" />
+  </Target>
+
+  <Target Name="BuildManaged">
+    <ItemGroup>
+      <_BuildMangedProjects Include="src\dirs.proj" />
```

Comment:
Inconsistent naming: why is this "dirs.proj" but native is "build-native.proj"?

Reply:
We could certainly rename dirs.proj to build-managed.proj but I didn't go that far yet. However I did want to make it clear that build-native.proj is intended to wrap the native build.